plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -252,48 +252,56 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
             }
             try {
                 outputLine = outputLine.trim();
+                Matcher m;
+                if (MYPY_MATCH_PATTERN1.matcher(outputLine).matches()) {
```

Comment:
You're doing multiple matches when you don't have to (regexp matching may be slow).

Also, you know from the matcher whether the col is available, so, you don't need use a raised exception to handle the flow in that case.

A better approach is probably something as:

```
Matcher m = MYPY_MATCH_PATTERN1.matcher(outputLine);
if(m.matches()){
   column = Integer.parseInt(outputLine.substring(m.start(3), m.end(3))) - 1
} else{
    m = MYPY_MATCH_PATTERN2.matcher(outputLine);
    if(m.matches()){
        column = 0;
    }else{
        m = null;
    }
}
```

and later instead of checking again if `m.matches()`, you can just do: 

```
if(m != null){
    ....
}
```